File: src/infrastructure/csvparser/CsvWriterImpl.cs
using System.Collections.Generic;
using System.Globalization;
using System.IO;
using core.fs.Shared.Adapters.CSV;
using CsvHelper;
using CsvHelper.Configuration;

namespace csvparser
{
    public class CsvWriterImpl : ICSVWriter
    {
        private readonly CsvConfiguration _config = new(CultureInfo.InvariantCulture)
        {
            HasHeaderRecord = true,
            Delimiter = ","
        };

        public string Generate<T>(IEnumerable<T> rows)
        {    
            using var writer = new StringWriter();
            using var csv = new CsvWriter(writer, _config);

            csv.WriteRecords(rows);

            return writer.ToString();
        }
    }
}

File: src/infrastructure/csvparser.fs/CsvWriterImpl.fs
namespace csvparser.fs

open System.IO
open System.Globalization
open CsvHelper
open CsvHelper.Configuration
open core.fs.Adapters.CSV

type CsvWriterImpl() =
    let config = 
        CsvConfiguration(CultureInfo.InvariantCulture,
            HasHeaderRecord = true,
            Delimiter = ","
        )
    
    interface ICSVWriter with
        member _.Generate<'T>(rows: seq<'T>) : string =
            use writer = new StringWriter()
            use csv = new CsvWriter(writer, config)
            
            csv.WriteRecords(rows)
            
            writer.ToString()
